src/message_conversions.rs

```diff
@@ -0,0 +1,530 @@
+            get_pinned_message_base(&vsomeip_msg).set_interface_version(interface_version);
+
+            let req_id = umsg.attributes.reqid.as_ref().ok_or_else(|| {
+                UStatus::fail_with_code(
+                    UCode::INVALID_ARGUMENT,
+                    "Missing id for Request message. Would be unable to correlate. Rejected.",
+                )
+            })?;
+            trace!(
+                "{} - Looking up req_id from UMessage in ME_REQUEST_CORRELATION, req_id: {}",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_UMSG_TO_VSOMEIP_MSG,
+                req_id.to_hyphenated_string()
+            );
+            let mut me_request_correlation = ME_REQUEST_CORRELATION.write().await;
+            let Some(request_id) = me_request_correlation.remove(req_id) else {
+                return Err(UStatus::fail_with_code(
+                    UCode::NOT_FOUND,
+                    format!("Corresponding SOME/IP Request ID not found for this Request UMessage's reqid: {}",
+                            req_id.to_hyphenated_string()),
+                ));
+            };
+            trace!(
+                "{} - Found correlated request_id: {}",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_UMSG_TO_VSOMEIP_MSG,
+                request_id
+            );
+            let (client_id, session_id) = split_u32_to_u16(request_id);
+            get_pinned_message_base(&vsomeip_msg).set_client(client_id);
+            get_pinned_message_base(&vsomeip_msg).set_session(session_id);
+            trace!(
+                "{} - request_id: {} client_id: {} session_id: {}",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_UMSG_TO_VSOMEIP_MSG,
+                request_id,
+                client_id,
+                session_id
+            );
+            let ok = {
+                if let Some(commstatus) = umsg.attributes.commstatus {
+                    let commstatus = commstatus.enum_value_or(UCode::UNIMPLEMENTED);
+                    commstatus == UCode::OK
+                } else {
+                    false
+                }
+            };
+            let payload = {
+                if let Some(bytes) = umsg.payload.clone() {
+                    bytes.to_vec()
+                } else {
+                    Vec::new()
+                }
+            };
+            let mut vsomeip_payload =
+                make_payload_wrapper(get_pinned_runtime(runtime_wrapper).create_payload());
+            set_data_safe(get_pinned_payload(&vsomeip_payload), &payload);
+            set_message_payload(&mut vsomeip_msg, &mut vsomeip_payload);
+            if ok {
+                get_pinned_message_base(&vsomeip_msg).set_return_code(vsomeip::return_code_e::E_OK);
+                get_pinned_message_base(&vsomeip_msg).set_message_type(message_type_e::MT_RESPONSE);
+            } else {
+                // TODO: Perform mapping from uProtocol UCode contained in commstatus into vsomeip::return_code_e
```

Comment:
Reminder to self to update this as appropriate with mapping outlined in `Table 5. UCode to SOME/IP Error Code Mapping` from [SOME/IP spec for uProtocol](https://github.com/eclipse-uprotocol/up-spec/blob/main/up-l1/someip.adoc#225-communication-status).